src/frontend/apps/e2e/__tests__/app-impress/common.ts

```diff
@@ -238,3 +238,39 @@ export const mockedAccesses = async (page: Page, json?: object) => {
     }
   });
 };
+
+// language helper
+export const TestLanguage = {
+  English: {
+    label: 'English',
+    expectedLocale: ['en-us'],
+  },
+  French: {
+    label: 'Français',
+    expectedLocale: ['fr-fr'],
+  },
+} as const;
+
+type TestLanguageKey = keyof typeof TestLanguage;
+type TestLanguageValue = (typeof TestLanguage)[TestLanguageKey];
+
+export async function waitForLanguageSwitch(
+  page: Page,
+  lang: TestLanguageValue,
+) {
+  const header = page.locator('header').first();
+  await header.getByRole('combobox').click();
+
+  const [response] = await Promise.all([
+    page.waitForResponse(
+      (resp) =>
+        resp.url().includes('/user') && resp.request().method() === 'PATCH',
+    ),
+    header.getByRole('option', { name: lang.label }).click(),
+  ]);
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const updatedUserResponse = await response.json();
+
+  expect(lang.expectedLocale).toContain(updatedUserResponse.language);
+}
```

Comment:
This part seems to be used only in `language.spec.ts`, I would let it inside this test so, at least until we need this part in another test.

Reply:
Agree, if language is involved, we should instead put the test in `language.spec.ts`